### ISSUE TITLE
Build v2.1.0.4 failure related to EC_curve_nist2nid

### ISSUE THREAD

**Issue by NoAgendaIT** [Gulden/gulden-official#165]
This is a Funtoo Linux machine, x64, with gcc 7.4.1, glibc 2.27.

I have pulled gulden from the github repository, checked out tag v2.1.0.4, and I have followed the build instructions and checked the dependencies that are appropriate for this distribution.

The system's qtnetwork is version 5.11.2, openssl is 1.0.2q.

I ran ./autogen, followed by cd depends, followed by make.

The compilation fails with the following error:

```
linking ../../lib/libQt5Network.so.5.7.1
/usr/lib/gcc/x86_64-pc-linux-gnu/7.4.1/../../../../x86_64-pc-linux-gnu/bin/ld: .obj/qsslsocket_openssl_symbols.o: in function `q_EC_curve_nist2nid(char const*)':
qsslsocket_openssl_symbols.cpp:(.text+0xb61): undefined reference to `EC_curve_nist2nid'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:519: ../../lib/libQt5Network.so.5.7.1] Error 1
make[2]: Leaving directory '/docs/markvd/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/qt/5.7.1-243acac7bae/qtbase/src/network'
make[1]: *** [Makefile:214: sub-network] Error 2
make[1]: Leaving directory '/docs/markvd/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/qt/5.7.1-243acac7bae/qtbase/src'
make: *** [funcs.mk:244: /docs/markvd/src/gulden-official/depends/work/build/x86_64-pc-linux-gnu/qt/5.7.1-243acac7bae/qtbase/.stamp_built] Error 2
```

**Comment by mjmacleod** [comment]
I'd recommend to try with 'make NO_QT=1' for the depends directory. This should skip building qt, and prefer the system qt when you configure for the main app.

I'd try debug further but 2.1.0.5 bumps qt to a newer version anyway, so might be a waste of time to look into it otherwise.

**Comment by NoAgendaIT** [comment]
Alright, thanks again @mjmacleod. If this happens again on v2.1.0.5, I'll reopen the bug.

**Comment by mjmacleod** [comment]
Okay, 2.1.0.5 is tagged, please do let me know if you run into any further build issues with it so we can clear them up.

My x86_64 gentoo dev machine is building it without issue, and funtoo being similar the same should apply your side (in theory) but of course in reality differences may crop up.

**Comment by NoAgendaIT** [comment]
Okay. Well, I have checked the tag and I ran 'git clean -fdx' to make it the same as the source. Ran autogen.sh and went into the depends dir. Now, when I run 'make':

```
 % make
/bin/sh ./config.status --recheck
/bin/sh: ./config.status: No such file or directory
```
I guess I need to run configure first? Maybe the install steps changed, please let me know.

**Comment by mjmacleod** [comment]
Install steps have not changed. Configure can only be run after the depends dir is built and shouldn't affect that directory in any way.

I can't think of any obvious reason it would be looking for config.status inside the depends directory, the makefile there doesn't use it (or autotools in general).
The various dependencies it builds (like boost) do but then I'd expect to see a lot more logging before the error.

What shell are you running these commands in?

**Comment by NoAgendaIT** [comment]
ZSH. But I must have done something wrong. I re-cloned the repo and make runs properly now. Apologies, it was a bit late last night. :-)

**Comment by NoAgendaIT** [comment]
Ok. Compilation succeeded. I do seem to have another problem now: I started Gulden (QT), created a new wallet, let it sync for a bit and then exited. When I started it back up, I got the following:

```
Gulden: ../../src/key.cpp:307: void ECC_Start(): Assertion `secp256k1_context_sign == NULL' failed.
[1]    9716 abort      src/qt/Gulden
```

It won't start up again. I could try to let it fully sync, but shouldn't it continue where it left off?

I'll refile this in a separate bug if you want me to @mjmacleod thanks.
